services/debug/api.go

```diff
@@ -0,0 +1,63 @@
+package debug
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+)
+
+var (
+	postTimeout = 60 * time.Second
+	errTimeout  = errors.New("timed out waiting for the envelope to be sent")
+	errExpired  = errors.New("envelope expired before being sent")
+)
+
+// PublicAPI represents a set of APIs from the `web3.debug` namespace.
+type PublicAPI struct {
+	s *Service
+}
+
+// NewAPI creates an instance of the debug API.
+func NewAPI(s *Service) *PublicAPI {
+	return &PublicAPI{s: s}
+}
+
+// PostSync sends an envelope through shhext_post and waits until the related
+// envelope event is sent.
+func (api *PublicAPI) PostSync(ctx context.Context, req whisper.NewMessage) (hash hexutil.Bytes, err error) {
+	hash, err = api.s.p.Post(ctx, req)
+	if err != nil {
+		return
+	}
+	err = api.waitForHash(hash)
```

Comment:
Should we pass the context to `waitForHash` as well? Or create `context.WithTimeout(ctx, postTimeout)` and pass that?

Reply:
Agree with funneling the context to `waitForHash`